### PR TITLE
cmake: sysbuild: signing: Fix DirectXIP signing with confirmation

### PR DESCRIPTION
**Pull request by nordicjm**
Adds the confirm flag when using imgtool to make signed hex files be confirmed

NCSDK-24020

### CODE REVIEW

**Comment by kapi-no** [comment]
Thanks, @nordicjm! This is a much-appreciated change in our team. Added people from my team so they can take a look and validate the change in our apps that use the direct-xip mode.

File: cmake/sysbuild/image_signing.cmake [hunk 1]
```diff
@@ -75,6 +75,10 @@ function(zephyr_mcuboot_tasks)
     else()
       set(imgtool_rom_command --rom-fixed @PM_MCUBOOT_PRIMARY_ADDRESS@)
     endif()
+
+    set(imgtool_directxip_hex_command --confirm)
```
Comment (kapi-no) [inline cmake/sysbuild/image_signing.cmake:79]: Shouldn't we only mark the image as confirmed for the `CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT` bootloader mode?

As far as I know, the `CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` variant does not use the confirmation mechanism.

Comment (de-nordic) [reply cmake/sysbuild/image_signing.cmake:79]: Confirm should have no effect on DIRECT_XIP without revert, as the order of execution is to pick the highest slot with highest version. Confirm is only used by REVERT to mark tried image to stay.

Comment (nvlsianpu) [reply cmake/sysbuild/image_signing.cmake:79]: @de-nordic confirm - it doesn't affect that mode at all.

**Review by gchwier — APPROVED** [review]
Looks good, and regression works.
I've removed `CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS` from our internal tests and tests for direct Xip and Direct Xip with revert are working:
https://jenkins-ncs.nordicsemi.no/job/latest/job/night/job/test-sdk-mcuboot/job/grch_rework_tests/20/testReport/
Legacy tests with child images still have this Kconfig set.

**Review by de-nordic — APPROVED** [review]

**Pull request merged**